Cabal/Distribution/Version.hs

```diff
@@ -387,15 +388,19 @@ orEarlierVersion v = UnionVersionRanges (ThisVersion v) (EarlierVersion v)
 -- > = withinRange v' vr1 || withinRange v' vr2
 --
 unionVersionRanges :: VersionRange -> VersionRange -> VersionRange
-unionVersionRanges = UnionVersionRanges
+unionVersionRanges (UnionVersionRanges a b) c =
+    unionVersionRanges a (unionVersionRanges b c)
+unionVersionRanges a b = UnionVersionRanges a b
 
 -- | The version range @vr1 && vr2@
 --
 -- >   withinRange v' (intersectVersionRanges vr1 vr2)
 -- > = withinRange v' vr1 && withinRange v' vr2
 --
 intersectVersionRanges :: VersionRange -> VersionRange -> VersionRange
-intersectVersionRanges = IntersectVersionRanges
+intersectVersionRanges (IntersectVersionRanges a b) c =
+    intersectVersionRanges a (intersectVersionRanges b c)
```

Comment:
Can the result of the second call also be a `IntersectVersionRanges`?

Reply:
I removed this change